pkg/api/types.go

```diff
@@ -256,6 +258,19 @@ type SecretVolumeSource struct {
 	Target ObjectReference `json:"target"`
 }
 
+// NFS represents and NFS Mount that lasts the lifetime of a pod
+type NFSVolumeSource struct {
+	// Server is the hostname or ip address of the NFS server
+	Server string `json:"server"`
+
+	// SourcePath is the exported NFS share
+	SourcePath string `json:"sourcePath"`
+
+	// MountOptions is a comma delimited list of mount options. The available options are listed on the mount man page as arguments to -o.
+	// As an example: noatime,nodev,nosuid,ro
+	MountOptions string `json:"mountOptions"`
```

Comment:
The rationale for expanding into a slice is that "a comma delimited string" is a leaky abstraction - it shows the implementation through the API.  You might as well ask for a bitwise-OR'ed set of MS_* flags.  I'd rather not offer generic options at all - that means we have to handle the user doing something really invalid.  Could we not offer options and expand this into the discrete flags we want?  Why not?

I went to re-look at iSCSI and maybe it was not that PR that made the suggestion.  Where was it?  I'll yield on this one if you and @rootfs and @markturansky all think comma-separated list is actually a better API.  @bgrant0607 in case he wants to weigh in as API overseer.  We do have to define what happens if the user specifies illegal or nonsensical options.

As for readOnly, it's the "offer discrete flags" model vs "offer arbitrary opaque options" model.  If we go down the path of opaque flags, we should get rid of iscsi.ReadOnly, too.  There's value in multiple plugins using the same basic model for these things, and readOnly feels like a common enough flag that it makes sense to call out.